### PR TITLE
feat(sdk/hide-sensitive-data): Add option to hide sensitive data

### PR DESCRIPTION
**Pull request by gagantrivedi**
Add option to hide "description" from the response of sdk endpoints

Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Add option to hide "description" from the response of sdk endpoints
## How did you test this code?

Adds unit test cases

### CODE REVIEW

File: api/features/serializers.py [hunk 1]
```diff
@@ -222,6 +225,15 @@ class Meta:
         writeonly_fields = ("initial_value", "default_enabled")
 
 
+class SDKFeatureSerializer(HideSensitiveFieldsSerializerMixin, FeatureSerializer):
+    sensitive_fields = (
```
Comment (gagantrivedi) [inline api/features/serializers.py:229]: All of them are not really sensitive fields, but I can't think of a better name

File: api/environments/identities/views.py [hunk 2]
```diff
@@ -247,7 +249,7 @@ def post(self, request):
         # trait values are serialized correctly
         response_serializer = IdentifyWithTraitsSerializer(
             instance=instance,
-            context={"identity": instance.get("identity")},  # todo: improve this
+            context=self._get_serializer_context(instance.get("identity")),
```
Comment (matthewelwell) [inline api/environments/identities/views.py]: I guess the alternative here is to do something like: 

```python
self.identity = instance.get("identity") 
```

in the code above and then just use `self.identity` in the `get_serializer_context()` method so we don't have to create this, slightly confusing, `_get_serializer_context` method. What do you think?

**Review by matthewelwell — COMMENTED** [review]
Some things that I've considered while reviewing this PR: 

 1. Should we just expose a field that allows people to define the fields that they want to hide? Maybe this is dangerous as it would allow them to break things? 
 2. Should we remove them from the response or just null / blank them out? (to avoid breaking changes)

**Comment by gagantrivedi** [comment]
> Some things that I've considered while reviewing this PR:
> 
>     1. Should we just expose a field that allows people to define the fields that they want to hide? Maybe this is dangerous as it would allow them to break things?
> 
>     2. Should we remove them from the response or just null / blank them out? (to avoid breaking changes)

Re 1 yeah, I don't really think that's a good idea, maybe we should have the button that says something like `use edge like response`?

**Comment by matthewelwell** [comment]
> Re 1 yeah, I don't really think that's a good idea, maybe we should have the button that says something like use edge like response?

Yeah, I guess we can be careful with the wording the UI. Perhaps make sure that we provide a big warning saying 'THIS WILL CAUSE A BREAKING CHANGE IN THE API (but won't break any Flagsmith SDKs)'

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**